src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -96,7 +97,9 @@ internal abstract class PEModuleBuilder<TCompilation, TSourceModuleSymbol, TAsse
         {
             Debug.Assert(sourceModule != null);
             Debug.Assert(serializationProperties != null);
+            Debug.Assert(debugEntryPoint == null || IsSourceDefinition(debugEntryPoint, sourceModule));
 
+            _debugEntryPoint = (Cci.IMethodDefinition)debugEntryPoint;
```

Comment:
Ah, yes, found it. We haven't translated the PE entry point either. Translating both now.